modules/network/src/main/java/org/apache/ignite/network/internal/netty/NettyClient.java

```diff
@@ -0,0 +1,141 @@
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.handler.stream.ChunkedWriteHandler;
+import java.net.InetSocketAddress;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiConsumer;
+import org.apache.ignite.network.message.MessageSerializationRegistry;
+import org.apache.ignite.network.message.NetworkMessage;
+
+/**
+ * Netty client channel wrapper.
+ */
+public class NettyClient {
+    /** Serialization registry. */
+    private final MessageSerializationRegistry serializationRegistry;
+
+    /** Destination host. */
+    private final String host;
+
+    /** Destination port. */
+    private final int port;
+
+    /** Future that resolves when client channel is opened. */
+    private final CompletableFuture<NettySender> clientFuture = new CompletableFuture<>();
+
+    /** Client channel. */
+    private SocketChannel channel;
+
+    public NettyClient(
+        String host,
+        int port,
+        MessageSerializationRegistry serializationRegistry
+    ) {
+        this.host = host;
+        this.port = port;
+        this.serializationRegistry = serializationRegistry;
+    }
+
+    /**
+     * Start client.
+     *
+     * @return Future that resolves when client channel is opened.
+     */
+    public CompletableFuture<NettySender> start(Bootstrap bootstrap) {
+        bootstrap.connect(host, port).addListener((ChannelFutureListener) connect -> {
+            this.channel = (SocketChannel) connect.channel();
+
+            if (connect.isSuccess())
+                clientFuture.complete(new NettySender(channel, serializationRegistry));
+            else
+                clientFuture.completeExceptionally(connect.cause());
+        });
+
+        return clientFuture;
+    }
+
+    /**
+     * @return Client start future.
+     */
+    public CompletableFuture<NettySender> sender() {
+        return clientFuture;
+    }
+
+    /**
+     * Stop client.
+     */
+    public void stop() {
+        this.channel.close().awaitUninterruptibly();
+    }
+
+    /**
+     * @return {@code true} if client failed to connect to remote server, {@code false} otherwise.
+     */
+    public boolean failedToConnect() {
+        return clientFuture.isCompletedExceptionally();
+    }
+
+    /**
+     * @return {@code true} if client lost connection, {@code false} otherwise.
+     */
+    public boolean isDisconnected() {
+        return channel != null && !channel.isOpen();
+    }
+
+    /**
+     * Creates {@link Bootstrap} for clients, providing channel handlers and options.
+     *
+     * @param eventLoopGroup Event loop group for channel handling.
+     * @param serializationRegistry Serialization registry.
+     * @param messageListener Message listener.
+     * @return Bootstrap for clients.
+     */
+    public static Bootstrap createBootstrap(
```

Comment:
I'm not sure if the `ConnectionManager` should configure client's implementation details

Reply:
I'm not sure if this method can be considered as client implementation details, since the Bootstrap is created in the ConnectionManager and then passed back into the client instance. So the actual client does not care about the particular Bootstrap implementation, since it is injected through the constuctor

Reply:
Ok then